pkg/service/healthcheck/service.go

```diff
@@ -399,6 +409,12 @@ func (s *Service) nodeInfo(ctx context.Context, clusterID uuid.UUID, host string
 	if ni, ok := s.nodeInfoCache[key]; ok && now.Before(ni.Expires) {
 		return ni, nil
 	}
+
+	c, err := s.clusterProvider(ctx, clusterID)
+	if err != nil {
+		return nodeInfo{}, err
+	}
```

Comment:
This is just a general approach comment.

Now i see that we are storing parts of TLS in 2 separate places:
- in `secrets` table - TLS cert and private key files
- in `cluster` table - `--force-tls-disabled`, `--force-non-ssl-session-port`

This results in the need of passing `ProviderFunc` into `healthcheck` logic.
What do you think about storing those information in the same place (e.g. add it to `secrets` table)? We always want to retrieve all or none of them, so it would be easier / more coherent this way.
What is your opinion on that? (btw this is the last moment to perhaps change this decision before new TLS logic is released)

Reply:
But isn't it logical ? 
In `secrets` table we store sensitive information like certificate and key.
In `cluster` we store cluster configuration.

I would prefer to not move cluster configuration to secrets.